vlog4j-client/src/main/java/org/vlog4j/client/picocli/VLog4jClientMaterialize.java

```diff
@@ -0,0 +1,161 @@
+
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(name = "materialize", description = "Execute the chase and store the literal's extensions")
+public class VLog4jClientMaterialize implements Runnable {
+
+	@Option(names = "--rule-file", description = "Rule file(s) in rls syntax", required = true)
+	private List<String> ruleFiles = new ArrayList<>();
+
+//	we need support to parseInto from gral files
+//	@Option(names = "--graal-rule-file", description = "Rule file(s) in graal syntax", required = true)
+//	private List<String> graalRuleFiles = new ArrayList<>();
+
+	// The value for annotation attribute CommandLine.Option.description must be a
+	// constant expression
+
+	@Option(names = "--log-level", description = "Log level of VLog (c++ library). One of: DEBUG, INFO, WARNING (default), ERROR.", required = false)
+	LogLevel logLevel = LogLevel.WARNING;
+
+	@Option(names = "--log-file", description = "Log file of VLog (c++ library). VLog will log to the default system output by default", required = false)
+	private String logFile;
+
+	@Option(names = "--chase-algorithm", description = "Chase algorithm. RESTRICTED_CHASE (default) or SKOLEM_CHASE.", required = false)
+	private Algorithm chaseAlgorithm = Algorithm.RESTRICTED_CHASE;
+
+	@Option(names = "--timeout", description = "Timeout in seconds. Infinite by default", required = false)
+	private int timeout = 0;
+
+	@Option(names = "--query", description = "Positive not-ground Literals to query after materialization in rls syntax. Vlog4jClient will print the size of its extension", required = true)
+	private List<String> queryStrings = new ArrayList<>();
+
+// TODO SaveModel
+//	/* group to save results */
+//	@ArgGroup(exclusive = false)
+//	private SaveModel saveModel;
+
+	@ArgGroup(exclusive = false)
+	private SaveQueryResult saveQueryResult = new SaveQueryResult();
+
+	/* group to print results */
+	@ArgGroup(exclusive = false)
+	private PrintQueryResults printQueryResult = new PrintQueryResults();
+
+	@Override
+	public void run() {
+		ExamplesUtils.configureLogging();
+		
+		System.out.println(queryStrings);
+
+//		if (saveModel.check() & saveQueryResult.check() & printQueryResult.check()) {
+//			saveModel.prepare();
+		if (saveQueryResult.checkConfiguration() & printQueryResult.checkConfiguration()) {
+			saveQueryResult.prepare();
+		} else {
+			System.exit(1);
+		}
+
+		System.out.println("Configuration:");
+
+		/* Configure rules */
+		final KnowledgeBase kb = new KnowledgeBase();
+		for (String ruleFile : ruleFiles) {
+			try {
+				RuleParser.parseInto(kb, new FileInputStream(ruleFile));
+				System.out.println("  --rule-file: " + ruleFile);
+			} catch (FileNotFoundException e) {
+				System.out.println("File not found: " + ruleFile + ". " + e.getMessage());
+			} catch (ParsingException e) {
+				System.out.println("Failed to parse rule file: " + ruleFile + ". " + e.getMessage());
+			}
+		}
```

Comment:
Do we want to continue reasoning if we encounter an exception while reading rules?